Mathlib/CategoryTheory/Sites/CartesianClosed.lean

```diff
@@ -20,6 +20,9 @@ open CategoryTheory Limits
 
 variable {C : Type*} [Category C] (J : GrothendieckTopology C) (A : Type*) [Category A]
 
-instance [HasSheafify J A] [HasFiniteProducts A] [CartesianClosed (Cᵒᵖ ⥤ A)] :
+instance [HasSheafify J A] [ChosenFiniteProducts A] : ChosenFiniteProducts (Sheaf J A) :=
+  reflectiveChosenFiniteProducts (sheafToPresheaf _ _)
```

Comment:
Before we proceed with this refactor PR, I think we should get a better `ChosenFiniteProducts` instance on sheaves. First, if `A` has chosen finite products, then functor categories to `A` also have chosen finite products. Then, as a fullsubcategory stable by finite products, the category of sheaves with values `A` should have an induced chosen finite products structure. As `Sheaf` is an adhoc structure (rather than `FullSubcategory`), we need to define a specific `ChosenFiniteProducts` instance: the only mathematically relevant theorems is that if we have two sheaves, then the the chosen binary product of the underlying presheaves is a sheaf (use `isSheaf_of_isLimit` in `Sites.Limits`)

Reply:
I would think this construction should go in a separate PR.

Reply:
This construction is in #18633.